Add `Matrix4.createPerspective()`

bwkam opened the pull request:
I didn't find a createPerspective function (for creating perspective views) so I tried to make one, but I don't really know if that will work.

bwkam committed:
add createPerspective

I didn't find a createPerspective function so I tried to make one, but I don't really know if that will work.

bwkam committed:
forgot something

bwkam committed:
add doc

bwkam committed:
I hope that's correct 

added zero to one mode, and -one to one mode variants

bwkam committed:
fix missing import

bwkam committed:
I keep forgetting shit

bwkam committed:
change to exceptions

bwkam commented:
Can someone re-run the CI?

player-03 commented on src/lime/math/Matrix4.hx:
```diff
@@ -378,6 +379,52 @@ abstract Matrix4(Float32Array) from Float32Array to Float32Array
+			this[12] = -1.0;
+			this[15] = -(zFar * zNear) / (zFar - zNear);
+		} else {
+			throw new haxe.Exception("aspect greater than epsilson for Float");
```
Don't think this works with Haxe 3. Maybe just throw the string without wrapping it.

player-03 replied:
Also without the `new` keyword. :)

bwkam replied:
forgot about that lol

bwkam committed:
Update Matrix4.hx

bwkam committed:
remove `new`

bwkam committed:
remove period

bwkam commented:
Does that look good?

bwkam committed:
Merge branch 'openfl:develop' into patch-1

bwkam committed:
final changes

bwkam commented:
![Peek 2023-05-10 09-05](https://github.com/openfl/lime/assets/91009118/54ffdf7b-e078-4b82-b525-df7684bcc194)

Here is a [demo](https://github.com/bwkam/learnopengl-haxelime/tree/main/Getting-Started/Coordinate-Systems/Rotating3DContainer)
Thanks to [@47rooks](https://github.com/47rooks) for the modifications.

player-03 committed:
Follow formatting conventions.

player-03 committed:
Avoid using `Math.pow()`.

Haxe is unable to optimize this function away. Also, it seems to be unable to return numbers smaller than 1e-7, so that's what I used in its place. Smaller constants can work, you just can't get them from `Math.pow()`.

player-03 committed:
Fill in missing entries.

As per all the other "create" functions.

player-03 committed:
Use guard clause for better readability.

player-03 committed:
Fix duplicate assignment.

player-03 committed:
Make `createPerspective()` public.
